SU-SWS/decanter
# Grids

# READY FOR REVIEW

# Summary
- Slight re-factoring to our grid systems & an addition of many mixins
- A couple dozen layout examples using the css grid based grid system
- Started to create clearer separation of what is a css grid and what is a flex grid with new naming conventions
- Removed previous page layout examples
- Upgraded grunt sass version to the latest 
- Added new and improved mixins for the flex grid

I went through an excercise in order to test out the Grids that are in our framework and how it is to use them. My exercise was to re-create a bunch of simple to complex layouts that I had previously created using Decanter V1 and neat's grid system. At the start I had all the tools I needed to create a few simple layouts but I was annoyed by how much code I had to write and remember to account for all of the grid breakpoints and scaling margins and gutters. This fustration lead to the creation of many mixins to help expidite and simplify the creation of responsive and flexible grid systems. I cherry picked a few mixins from internet searches as well as some from neat and reverse engineered them back in to our framework. At the end of my exercise I had a huge collection of mixins and some really clean and stripped down semantic templates. 

There is still a significant amount of clean up work to be done to refactor some of our components and to ensure cross browser compatibility but this is should be a significant start.

# Needed By (Date)
- ?

# Urgency
- None

# Steps to Test

1. Will schedule a first pass meeting to go over the changes in detail with you.
2. In the mean time please check out this branch
3. Run `npm install`
4. Compile the styleguide
5. Open a browser and navigate to the layouts section to see the new layouts
6. Start reading some of the code in the layout examples to see the general approach.

# Affected Projects or Products
- Pretty much everything

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
